app/src/test/java/com/codesoom/assignment/repository/ProductRepositoryTest.java

```diff
@@ -0,0 +1,171 @@
+package com.codesoom.assignment.repository;
+
+import com.codesoom.assignment.domain.Product;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.dao.EmptyResultDataAccessException;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.*;
+
+@DataJpaTest
+@DisplayName("ProductRepository JPA 테스트")
+class ProductRepositoryTest {
+
+    private final String name = "My toy";
+    private final String maker = "toy shop";
+    private final int price = 15000;
+    private final String imageUrl = "toy.jpg";
+
+    @Autowired
+    private ProductRepository productRepository;
+
+    @AfterEach
+    void clear() {
+        productRepository.deleteAll();
+    }
+
+    @Nested
+    @DisplayName("save 메소드는")
+    class Describe_save {
+        private final Product product = new Product(name, maker, price, imageUrl);
+
+        @Test
+        @DisplayName("장난감을 저장하고 반환한다")
+        void it_returns_saved_product() {
+            Product savedProduct = productRepository.save(product);
+
+            assertThat(savedProduct.getId()).isNotNull();
+            assertThat(savedProduct.getName()).isEqualTo(name);
+            assertThat(savedProduct.getMaker()).isEqualTo(maker);
+            assertThat(savedProduct.getPrice()).isEqualTo(price);
+            assertThat(savedProduct.getImageUrl()).isEqualTo(imageUrl);
+        }
+    }
+
+    @Nested
+    @DisplayName("findAll 메소드는")
+    class Describe_findAll {
+
+        @Nested
+        @DisplayName("만약 저장된 장난감이 없으면")
+        class Context_without_product {
+
+            @Test
+            @DisplayName("빈 목록을 반환한다")
+            void it_returns_empty_list() {
+                List<Product> products = productRepository.findAll();
+
+                assertThat(products).isEmpty();
+            }
+        }
+
+        @Nested
+        @DisplayName("만약 저장된 장난감이 있으면")
+        class Context_with_product {
+            private final int productSize = 3;
+
+            @BeforeEach
+            void setUp() {
+                for (int i = 0; i < productSize; i++) {
+                    Product product = new Product(name, maker, price, imageUrl);
+                    productRepository.save(product);
+                }
+            }
+
+            @Test
+            @DisplayName("모든 장난감을 반환한다")
+            void it_returns_products() {
+                List<Product> products = productRepository.findAll();
+
+                assertThat(products).hasSize(productSize);
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("findById 메소드는")
+    class Describe_findById {
+
+        @Nested
+        @DisplayName("만약 조회 가능한 id가 주어지면")
```

Comment:
같은 의미를 갖는다면 조금 더 짧은 표현을 쓰는 방향으로 작성해 보세요.
```suggestion
        @DisplayName("조회 가능한 id가 주어지면")
```